### ISSUE TITLE
plymouth update causes splashscreen to cover graphical session

### ISSUE THREAD

**Issue by Informatic** [NixOS/nixpkgs#200861]
### Describe the bug
Using plymouth, gdm and sway.

Whenever major system update involving new plymouth version is applied, plymouth units are restarted at the end, which makes it cover already running graphical user session.

```
stopping the following units: [...], plymouth-quit-wait.service, plymouth-read-write.service, plymouth-start.service, power-profiles-daemon.service, rtkit-daemon.service, systemd-ask-password-plymouth.service, [...]
Warning: Stopping systemd-ask-password-plymouth.service, but it can still be activated by:
  systemd-ask-password-plymouth.path
NOT restarting the following changed units: [...]
activating the configuration...
setting up /etc...
restarting systemd...
reloading user units for informatic...
setting up tmpfiles
reloading the following units: dbus.service, firewall.service, reload-systemd-vconsole-setup.service
restarting the following units: nix-daemon.service, polkit.service, systemd-journald.service
starting the following units: [...], plymouth-quit-wait.service, plymouth-read-write.service, plymouth-start.service, power-profiles-daemon.service, rtkit-daemon.service, systemd-ask-password-plymouth.service, [...]
the following new units were started: [...]
```

### Steps To Reproduce
1. Enable plymouth, gdm and sway, log into the graphical session.
2. Update channel/nixpkgs version to one with newer plymouth build
3. plymouth units are restarted, vt is changed to vt1, and splashscreen is displayed until Ctrl-Alt-F2 is used to switch back to sway. 

### Expected behavior
Plymouth update should not cause splash screen to show up.

### Additional context
Using plymouth, gdm and sway, but this would probably affect plymouth-only (no graphical session) setups too.

I am assuming we should add `restartIfChanged = false;` to all relevant plymouth units.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.64, NixOS, 22.11 (Raccoon), 22.11.git.623710774a9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos-22.05, nixos17-17.09, nixos2105-21.05"`
 - channels(informatic): `"nixos-21.05-21.05, nixos2105-21.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

(228feb709b63a8f0a80b55d9248d8b05dfa70c2e with minor unrelated patch applied)

**Comment by Informatic** [comment]
Seems like adding these options help:
```
# Prevents splashscreen from showing during update
systemd.services.plymouth-start.restartIfChanged = false;

# Prevents plymouth-quit-wait hanging during switch
systemd.services.plymouth-quit-wait.restartIfChanged = false;
```

**Comment by Informatic** [comment]
Fixed by #199784
